modules/alloydb/main.tf

```diff
@@ -17,42 +17,60 @@
 locals {
   prefix      = var.prefix == null ? "" : "${var.prefix}-"
   is_regional = var.availability_type == "REGIONAL"
-  # secondary instance type is aligned with cluster type unless apply is targeting a promotion, in that
-  # case cluster will be 'primary' while instance still 'secondary'.
+  flags       = try(var.network_config.psa_config.enable_outbound_public_ip, false) ? merge(var.flags, { "password.enforce_complexity" = "on" }) : var.flags # Outbound Public IP requires password complexity to be enabled.
```

Comment:
Reformat as follows.
```terraform
  # Outbound Public IP requires password complexity to be enabled.
  flags = (
    try(var.network_config.psa_config.enable_outbound_public_ip, false) 
    ? merge(var.flags, { "password.enforce_complexity" = "on" }) 
    : var.flags 
  )
```

Does the API warn you about this? If yes, then the user should be the one taking care of this.

Reply:
API does want about this. I will remove auto-setting of this flag.